### PR TITLE
Instead of returning null on getServerVersion(), it returns now a version

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
I use NC10 as a very very old fallback.

Signed-off-by: tobiasKaminsky <tobias@kaminsky.me>

### CODE REVIEW

**Comment by AndyScherzinger** [comment]
:+1:

File: src/main/java/com/owncloud/android/ui/helpers/FileOperationsHelper.java [hunk 1]
```diff
@@ -866,9 +866,8 @@ public void setOpIdWaitingFor(long waitingForOpId) {
      */
     public boolean isVersionWithForbiddenCharacters() {
         if (mFileActivity.getAccount() != null) {
-            OwnCloudVersion serverVersion =
-                    AccountUtils.getServerVersion(mFileActivity.getAccount());
-            return (serverVersion != null && serverVersion.isVersionWithForbiddenCharacters());
+            OwnCloudVersion serverVersion = AccountUtils.getServerVersion(mFileActivity.getAccount());
+            return serverVersion.isVersionWithForbiddenCharacters();
```
Comment (mario) [inline src/main/java/com/owncloud/android/ui/helpers/FileOperationsHelper.java]: Please use: return AccountUtils.getServerVersion(mFileActivity.getAccount()).isVersionWithForbiddenCharacters();

File: src/main/java/com/owncloud/android/authentication/AccountUtils.java [hunk 2]
```diff
@@ -211,13 +213,13 @@ public static OwnCloudVersion getServerVersion(Account account) {
         return serverVersion;
     }
 
-    public static boolean hasSearchUsersSupport(Account account){
+    public static boolean hasSearchUsersSupport(Account account) {
         OwnCloudVersion serverVersion = getServerVersion(account);
-        return (serverVersion != null && serverVersion.isSearchUsersSupported());
+        return serverVersion.isSearchUsersSupported();
```
Comment (mario) [inline src/main/java/com/owncloud/android/authentication/AccountUtils.java]: Please use return getServerVersion(account).isSearchUsersSupported();

File: src/main/java/com/owncloud/android/authentication/AccountUtils.java [hunk 3]
```diff
@@ -211,13 +213,13 @@ public static OwnCloudVersion getServerVersion(Account account) {
...
+    public static boolean hasSearchUsersSupport(Account account) {
         OwnCloudVersion serverVersion = getServerVersion(account);
-        return (serverVersion != null && serverVersion.isSearchUsersSupported());
+        return serverVersion.isSearchUsersSupported();
     }
 
     public static boolean hasSearchSupport(Account account) {
         OwnCloudVersion serverVersion = getServerVersion(account);
-        return (serverVersion != null && serverVersion.isSearchSupported());
+        return serverVersion.isSearchSupported();
```
Comment (mario) [inline src/main/java/com/owncloud/android/authentication/AccountUtils.java]: Same as above comment.

File: src/main/java/com/owncloud/android/ui/activity/ShareActivity.java [hunk 4]
```diff
@@ -149,9 +149,9 @@ private int getAppropiatePermissions(ShareType shareType) {
             return OCShare.READ_PERMISSION_FLAG;    // minimum permissions
 
         } else if (isFederated) {
-            OwnCloudVersion serverVersion = com.owncloud.android.authentication.AccountUtils.
-                    getServerVersion(getAccount());
-            if (serverVersion != null && serverVersion.isNotReshareableFederatedSupported()) {
+            OwnCloudVersion serverVersion = com.owncloud.android.authentication.AccountUtils
+                    .getServerVersion(getAccount());
+            if (serverVersion.isNotReshareableFederatedSupported()) {
```
Comment (mario) [inline src/main/java/com/owncloud/android/ui/activity/ShareActivity.java]: Same as above.

File: src/main/java/com/owncloud/android/ui/fragment/OCFileListFragment.java [hunk 5]
```diff
@@ -1224,10 +1224,8 @@ private void updateLayout() {
         if (!mJustFolders) {
             updateFooter();
             // decide grid vs list view
-            OwnCloudVersion version = AccountUtils.getServerVersion(
-                    ((FileActivity) mContainerActivity).getAccount());
-            if (version != null && version.supportsRemoteThumbnails() &&
-                    isGridViewPreferred(mFile)) {
+            OwnCloudVersion version = AccountUtils.getServerVersion(((FileActivity) mContainerActivity).getAccount());
+            if (version.supportsRemoteThumbnails() && isGridViewPreferred(mFile)) {
```
Comment (mario) [inline src/main/java/com/owncloud/android/ui/fragment/OCFileListFragment.java]: Same as above.

File: src/main/java/com/owncloud/android/ui/helpers/FileOperationsHelper.java [hunk 6]
```diff
@@ -387,7 +387,7 @@ public void shareFileWithSharee(OCFile file, String shareeName, ShareType shareT
     public boolean isSharedSupported() {
         if (mFileActivity.getAccount() != null) {
             OwnCloudVersion serverVersion = AccountUtils.getServerVersion(mFileActivity.getAccount());
-            return (serverVersion != null && serverVersion.isSharedSupported());
+            return serverVersion.isSharedSupported();
```
Comment (mario) [inline src/main/java/com/owncloud/android/ui/helpers/FileOperationsHelper.java]: Same.

File: src/main/java/com/owncloud/android/ui/helpers/FileOperationsHelper.java [hunk 7]
```diff
@@ -554,7 +554,7 @@ public void setHideFileListingPermissionsToShare(OCShare share, boolean hideFile
         } else {
             OwnCloudVersion serverVersion = AccountUtils.getServerVersion(mFileActivity.getAccount());
 
-            if (serverVersion != null && serverVersion.isNotReshareableFederatedSupported()) {
+            if (serverVersion.isNotReshareableFederatedSupported()) {
```
Comment (mario) [inline src/main/java/com/owncloud/android/ui/helpers/FileOperationsHelper.java]: Same.

File: src/main/java/com/owncloud/android/ui/helpers/FileOperationsHelper.java (hunk 1 above)
Comment (mario) [inline src/main/java/com/owncloud/android/ui/helpers/FileOperationsHelper.java]: Same.

File: src/main/java/com/owncloud/android/utils/glide/HttpStreamFetcher.java [hunk 8]
```diff
@@ -55,13 +55,12 @@ public HttpStreamFetcher(String url) {
...
-        OwnCloudAccount ocAccount = new OwnCloudAccount(mAccount,
-                MainApp.getAppContext());
+        OwnCloudAccount ocAccount = new OwnCloudAccount(mAccount, MainApp.getAppContext());
         OwnCloudClient mClient = OwnCloudClientManagerFactory.getDefaultSingleton().
                 getClientFor(ocAccount, MainApp.getAppContext());
 
         OwnCloudVersion serverOCVersion = AccountUtils.getServerVersion(mAccount);
-        if (mClient != null && serverOCVersion != null) {
+        if (mClient != null) {
             if (serverOCVersion.supportsRemoteThumbnails()) {
```
Comment (mario) [inline src/main/java/com/owncloud/android/utils/glide/HttpStreamFetcher.java]: Same.

**Comment by tobiasKaminsky** [comment]
Changes done

File: src/main/java/com/owncloud/android/ui/activity/ShareActivity.java [hunk 9]
```diff
@@ -149,9 +148,8 @@ private int getAppropiatePermissions(ShareType shareType) {
             return OCShare.READ_PERMISSION_FLAG;    // minimum permissions
 
         } else if (isFederated) {
-            OwnCloudVersion serverVersion = com.owncloud.android.authentication.AccountUtils
-                    .getServerVersion(getAccount());
-            if (serverVersion.isNotReshareableFederatedSupported()) {
+            if (com.owncloud.android.authentication.AccountUtils
```
Comment (mario) [inline src/main/java/com/owncloud/android/ui/activity/ShareActivity.java]: Can we please do proper import here instead of doing this com.owncloud.... xD)

Comment (tobiasKaminsky) [reply src/main/java/com/owncloud/android/ui/activity/ShareActivity.java]: This is not possible as we already have another AccountUtils imported:
import com.owncloud.android.lib.common.accounts.AccountUtils;

So one of the two has to have the full import path.
com.owncloud.android.authentication.AccountUtils is used only 1 time, whereas the other is used a bit more.

Comment (AndyScherzinger) [reply src/main/java/com/owncloud/android/ui/activity/ShareActivity.java]: @mario due to Tobias' comment this full.name reference will have to stay :/

Comment (mario) [reply src/main/java/com/owncloud/android/ui/activity/ShareActivity.java]: @AndyScherzinger yup, sure.

**Comment by mario** [comment]
👍

**Pull request merged**